packages/lesswrong/components/posts/AllPostsPage.jsx

```diff
@@ -1,91 +1,102 @@
-    width: "100%",
-    fontStyle: "italic",
-    cursor: "pointer",
-    opacity: .8,
-    [theme.breakpoints.up('md')]: {
-      float: "right"
-    },
-    '&:hover': {
-      opacity:1,
-    },
-    '& svg': {
-      width: ".8em",
-      marginTop: -4,
-      marginRight: 3
-    }
-  },
-  checkbox: {
-    padding: 0,
-  },
-  checkboxChecked: {
-    // Tone down the material-UI default color to the shade old-material-UI was
-    // using, since the new, darker green doesn't fit the deemphasized position
-    // this element is in.
-    "& svg": {
-      color: "rgba(100, 169, 105, 0.7)",
-    }
+  allPostsContent: {
+    padding: theme.spacing.unit
   },
-  checkboxLabel: {
-    ...theme.typography.subheading,
-    marginLeft: 5
+  settingsIcon: {
+    paddingLeft: theme.spacing.unit,
+    paddingRight: theme.spacing.unit,
+    color: theme.palette.grey[400],
+    width:40,
+    cursor: "pointer"
   },
-  divider: {
-    margin:"10px 0 12px 82%",
-    borderBottom: "solid 1px rgba(0,0,0,.15)"
-  }
 });
 
 class AllPostsPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showSettings: (props.currentUser && props.currentUser.allPostsOpenSettings) || false,
+    };
+  }
 
-  state = { lowKarma: false }
-
-  renderTitle = () => {
-    const { classes } = this.props;
-    const { lowKarma } = this.state
-    const { PostsViews } = Components
-    return <div>
-      <PostsViews />
-      <div className={classes.divider} />
-      <Tooltip title="Show low karma posts" placement="right">
-        <div className={classes.titleSettings} onClick={(event) => this.setState({lowKarma: !lowKarma})}>
-          <Checkbox
-            classes={{root: classes.checkbox, checked: classes.checkboxChecked}}
-            checked={lowKarma}
-          />
-          <span className={classes.checkboxLabel}>
-            Low Karma
-          </span>
-        </div>
-      </Tooltip>
-    </div>
+  toggleSettings = () => {
+    const { currentUser, updateUser } = this.props
+    this.setState({showSettings: !this.state.showSettings})
+    if (currentUser) {
+      updateUser({
+        selector: { _id: currentUser._id},
+        data: {
+          allPostsOpenSettings: !this.state.showSettings,
+        },
+      })  
+    }
   }
 
   render() {
+    const { classes, currentUser, router } = this.props
+    const { showSettings } = this.state
+    const { AllPostsPageSettings, PostsList, SingleColumnSection, SectionTitle } = Components
+    const query = _.clone(router.location.query) || {}
 
-    const query = _.clone(this.props.router.location.query || {});
+    const currentView = query.view || (currentUser && currentUser.allPostsView) || "daily"
+    const currentFilter = query.filter || (currentUser && currentUser.allPostsFilter) || "all"
+    const currentShowLowKarma = (query.karmaThreshold && (query.karmaThreshold == MAX_LOW_KARMA_THRESHOLD)) || (currentUser && currentUser.allPostsShowLowKarma) || false
```

Comment:
Tripple equal, not double equal

Reply:
Also, the null check here is unnecessary, since if it's null, it can never be equal to the MAX_LOW_KARMA_THRESHOLD variable